kafka/producer/main.go

```diff
@@ -29,139 +31,106 @@ const eventHubNamespace = "EVENTHUB_NAMESPACE"
-					log.Printf("Event sending failed %s", err.Error())
-				}
-
-				// create the next batch we'll use for events, ensuring that we use the same options
-				// each time so all the messages go the same target.
-				tmpBatch, err := producerClient.NewEventDataBatch(context.Background(), newBatchOptions)
-
-				if err != nil {
-					log.Printf("Creating new batch failed: %s", err.Error())
-				}
-
-				batch = tmpBatch
-
-				// rewind so we can retry adding this event to a batch
-				i--
-			} else if err != nil {
-				log.Printf("Errored while adding events to batch: %s", err.Error())
-			}
+		err = batch.AddEventData(event, nil)
+		if err != nil {
+			log.Printf("Adding event data to batch failed: %s", err.Error())
 		}
 
-		// if we have any events in the last batch, send it
-		if batch.NumEvents() > 0 {
-			log.Println("sending events")
-			if err := producerClient.SendEventDataBatch(context.Background(), batch, nil); err != nil {
-				log.Printf("Error sending events: %s", err.Error())
-			}
+		if err := producerClient.SendEventDataBatch(context.Background(), batch, nil); err != nil {
+			log.Printf("Event sending failed %s", err.Error())
 		}
-		log.Println("Rest for 10 seconds before sending additional events.")
+
+		select {
+		case sig := <-signals:
+			log.Printf("Got signal: %v", sig)
+			return
+		default:
+		}
+
 		time.Sleep(time.Second * 1)
 	}
 }
 
-func createEventsForDemo() []*azeventhubs.EventData {
-	rand.Seed(time.Now().UnixNano())
-	randomId := rand.Intn(90000) + 10000
-	rawMessage := getEnv(msg)
-	value := fmt.Sprintf("Message Id %d: %s", randomId, rawMessage)
-	encryptedValue, err := encryptMessage(value, "path_to_public_key.pem")
+func createEventsForDemo() *azeventhubs.EventData {
+	eventId += 1
+	rawMessage := util.GetEnv(msg)
+	value := fmt.Sprintf("Message Id %d: %s", eventId, rawMessage)
+	encryptedValue, err := encryptMessage(value)
 	if err != nil {
-		log.Println("Encrypting message failed.")
-	}
-	return []*azeventhubs.EventData{
-		{
-			Body: []byte(encryptedValue),
-		},
-		{
-			Body: []byte(encryptedValue),
-		},
+		log.Printf("Encrypting message failed: %s", err.Error())
 	}
+	return &azeventhubs.EventData{Body: []byte(encryptedValue)}
 }
 
-func encryptMessage(plaintext string, publicKeyPath string) (string, error) {
+func encryptMessage(plaintext string) (string, error) {
 	var pubpem []byte
 	var err error
-	if pkey := getEnv("PUBKEY"); len(pkey) > 0 {
+	if pkey := util.GetEnv("PUBKEY"); len(pkey) > 0 {
 		pubpem = []byte(pkey)
 	}
-	if len(pubpem) == 0 {
-		pubpem, err = os.ReadFile(publicKeyPath)
-		if err != nil {
-			return "", fmt.Errorf("failed to read public key file %v", publicKeyPath)
-		}
-	}
 	block, _ := pem.Decode([]byte(pubpem))
 	key, err := x509.ParsePKIXPublicKey(block.Bytes)
 	if err != nil {
-		return "", fmt.Errorf("invalid public key: %v", err)
+		return "Invalid public key", fmt.Errorf("invalid public key: %v", err)
```

Comment:
what's the motivation for returning a string when there's an error (the motivation for the change below)? 
before: `return "", fmt.Errorf("invalid public key: %v", err)`
after: `return "Invalid public key", fmt.Errorf("invalid public key: %v", err)`

the first one makes more sense to me since it ensures that only valid results are returned

(same question about L129, L132)

Reply:
honestly it does not matter what we return because if the pubkey has issues, then the private key must have issues because these two things are generated together. Users won't be able to see the return messages anyway. I made this change because I hope that in case the private key does not gets released, users can see the error from producer on the consumer side.